tasks/CommandInjection.cpp

```diff
@@ -0,0 +1,83 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "CommandInjection.hpp"
+
+using namespace auv_control;
+
+CommandInjection::CommandInjection(std::string const& name)
+    : CommandInjectionBase(name)
+{
+}
+
+CommandInjection::CommandInjection(std::string const& name, RTT::ExecutionEngine* engine)
+    : CommandInjectionBase(name, engine)
+{
+}
+
+CommandInjection::~CommandInjection()
+{
+}
+
+bool CommandInjection::calcOutput()
+{
+    base::LinearAngular6DCommand output_command = merged_command;
+
+    if(_cmd_injection.connected())
+    {
+        base::LinearAngular6DCommand cmd_injection;
+        if(_cmd_injection.readNewest(cmd_injection) == RTT::NewData)
```

Comment:
> I would assume to get the latest sample would be exactly what I want to have here. Especially when replaying log data faster than the update hook is triggered.

Then you are assuming that you know what the system builder wants. This kind of choices should be made with data/buffer connections, not forced in the C++ code by calling readNewest. readNewest was and still is a bad idea.

> True, but since this happens in a calcOutput() callback that in this case always should write out a sample and return true it will be harder to read and understand the code.

Well, at least explicitely testing against the return value of read() allows you to remove the connected() test, which I guess would make the whole thing more readable. Extracting the apply-override-to-command into a separate method would then make the logic really explicit.